### ISSUE TITLE
Much slower execution of compileJava task after ugrade from 6.0-6.4.1 to 6.5-6.8 (nightly)

### ISSUE THREAD

**Issue by ve2m** [gradle/gradle#14720]
`compileJava` execution time is more than 3 times slower (~12s vs ~39s) when upgrading from 6.0-6.4.1 to 6.5-6.8 (nightly). 

### Expected Behavior
Performance should be more or less the same.

### Current Behavior
Execution time is more than 3 times slower.

### Context

I've tried to use the `--debug` option and it looks like the slower version has some problems with memory management.

**Slow version (Gradle 6.5):**

![image](https://user-images.githubusercontent.com/14804545/94796725-71658100-03df-11eb-88dc-18f855b6f02f.png)

```
2020-09-30T22:52:32.447+0200 [DEBUG] [org.gradle.api.internal.tasks.compile.NormalizingJavaCompiler] 2020-09-30T22:52:32.385+0200 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationExecutor] Build operation 'Compile Java for :compileJava' started
2020-09-30T22:52:32.449+0200 [INFO] [org.gradle.api.internal.tasks.compile.JdkJavaCompiler] [org.gradle.api.internal.tasks.compile.JdkJavaCompiler] ] Compiling with JDK Java compiler API.Compiling with JDK Java compiler API.

2020-09-30T22:52:35.622+02002020-09-30T22:52:35.622+0200 [LIFECYCLE] [org.gradle.process.internal.health.memory.DefaultMemoryManager] 
2020-09-30T22:52:35.622+0200 [DEBUG] [org.gradle.process.internal.health.memory.DefaultMemoryManager] 1705261465 memory requested, 433627136 free
2020-09-30T22:52:35.622+0200 [DEBUG] [org.gradle.workers.internal.WorkerDaemonExpiration] Will attempt to release 1626 of memory [LIFECYCLE] [org.gradle.process.internal.health.memory.DefaultMemoryManager] 
2020-09-30T22:52:35.622+0200 [DEBUG] [org.gradle.process.internal.health.memory.DefaultMemoryManager] 1705261465 memory requested, 433627136 free
2020-09-30T22:52:35.622+0200 [DEBUG] [org.gradle.workers.internal.WorkerDaemonExpiration] Will attempt to release 1626 of memory
2020-09-30T22:52:35.622+0200 [
2020-09-30T22:52:35.622+0200 [DEBUG] [org.gradle.process.internal.health.memory.DefaultMemoryManagerDEBUG] [org.gradle.process.internal.health.memory.DefaultMemoryManager] 1705261465 memory requested, 0 released, 433627136 free
] 1705261465 memory requested, 0 released, 433627136 free
2020-09-30T22:52:37.353+02002020-09-30T22:52:37.353+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire shared lock on daemon addresses registry.
2020-09-30T22:52:37.353+0200 [DEBUG] [ [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire shared lock on daemon addresses registry.
2020-09-30T22:52:37.353+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
2020-09-30T22:52:37.354+0200 [DEBUG] [2020-09-30T22:52:37.354+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
2020-09-30T22:52:37.355+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager
2020-09-30T22:52:37.355+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire shared lock on daemon addresses registry.
] Waiting to acquire shared lock on daemon addresses registry.
2020-09-30T22:52:37.355+0200 [DEBUG] [2020-09-30T22:52:37.355+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
2020-09-30T22:52:37.355+0200
2020-09-30T22:52:37.355+0200 [DEBUG] [ [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
2020-09-30T22:52:40.623+02002020-09-30T22:52:40.623+0200 [ [DEBUGDEBUG] [org.gradle.process.internal.health.memory.DefaultMemoryManager] 1705261465 memory requested, 469573632 free] [org.gradle.process.internal.health.memory.DefaultMemoryManager] 1705261465 memory requested, 469573632 free

2020-09-30T22:52:40.623+02002020-09-30T22:52:40.623+0200 [DEBUG] [ [DEBUG] [org.gradle.workers.internal.WorkerDaemonExpirationorg.gradle.workers.internal.WorkerDaemonExpiration] Will attempt to release 1626 of memory
] Will attempt to release 1626 of memory
2020-09-30T22:52:40.623+0200 [DEBUG] [2020-09-30T22:52:40.623+0200 [DEBUG] [org.gradle.process.internal.health.memory.DefaultMemoryManager] org.gradle.process.internal.health.memory.DefaultMemoryManager] 1705261465 memory requested, 0 released, 469573632 free1705261465 memory requested, 0 released, 469573632 free

2020-09-30T22:52:33.943+02002020-09-30T22:52:33.943+0200 [LIFECYCLE] [class org.gradle.internal.buildevents.TaskExecutionLogger] 
 [LIFECYCLE] [class org.gradle.internal.buildevents.TaskExecutionLogger] 
2020-09-30T22:52:33.943+02002020-09-30T22:52:33.943+0200 [LIFECYCLE] [ [LIFECYCLE] [class org.gradle.internal.buildevents.TaskExecutionLogger] class org.gradle.internal.buildevents.TaskExecutionLogger] > Task :compileJava
> Task :compileJava
2020-09-30T22:52:45.622+02002020-09-30T22:52:45.622+0200 [LIFECYCLE] [org.gradle.process.internal.health.memory.DefaultMemoryManager] 
2020-09-30T22:52:45.622+0200 [DEBUG] [org.gradle.process.internal.health.memory.DefaultMemoryManager] 1705261465 memory requested, 536088576 free
 [LIFECYCLE] [org.gradle.process.internal.health.memory.DefaultMemoryManager] 
2020-09-30T22:52:45.622+0200 [DEBUG] [org.gradle.process.internal.health.memory.DefaultMemoryManager] 1705261465 memory requested, 536088576 free
2020-09-30T22:52:45.622+02002020-09-30T22:52:45.622+0200 [DEBUG] [org.gradle.workers.internal.WorkerDaemonExpiration] Will attempt to release 1626 of memory
2020-09-30T22:52:45.622+0200 [DEBUG] [org.gradle.process.internal.health.memory.DefaultMemoryManager] 1705261465 memory requested, 0 released, 536088576 free
 [DEBUG] [org.gradle.workers.internal.WorkerDaemonExpiration] Will attempt to release 1626 of memory
2020-09-30T22:52:45.622+0200 [DEBUG] [org.gradle.process.internal.health.memory.DefaultMemoryManager] 1705261465 memory requested, 0 released, 536088576 free
2020-09-30T22:52:47.352+02002020-09-30T22:52:47.352+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire shared lock on daemon addresses registry.
 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire shared lock on daemon addresses registry.
2020-09-30T22:52:47.352+0200 [2020-09-30T22:52:47.352+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
Lock acquired on daemon addresses registry.
2020-09-30T22:52:47.352+02002020-09-30T22:52:47.352+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
2020-09-30T22:52:47.352+0200 [2020-09-30T22:52:47.352+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire shared lock on daemon addresses registry.
Waiting to acquire shared lock on daemon addresses registry.
2020-09-30T22:52:47.352+0200 [DEBUG] [2020-09-30T22:52:47.352+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
2020-09-30T22:52:47.353+0200 [2020-09-30T22:52:47.353+0200 [DEBUG] [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
Releasing lock on daemon addresses registry.
2020-09-30T22:52:50.624+02002020-09-30T22:52:50.624+0200 [DEBUG] [org.gradle.process.internal.health.memory.DefaultMemoryManager] 1705261465 memory requested, 533757952 free
2020-09-30T22:52:50.624+0200 [DEBUG] [org.gradle.workers.internal.WorkerDaemonExpiration] Will attempt to release 1626 of memory
 [DEBUG] [org.gradle.process.internal.health.memory.DefaultMemoryManager] 1705261465 memory requested, 533757952 free
2020-09-30T22:52:50.624+0200 [DEBUG] [org.gradle.workers.internal.WorkerDaemonExpiration] Will attempt to release 1626 of memory
2020-09-30T22:52:50.624+0200 [DEBUG] [org.gradle.process.internal.health.memory.DefaultMemoryManager] 1705261465 memory requested, 0 released, 533757952 free
2020-09-30T22:52:50.624+0200 [DEBUG] [org.gradle.process.internal.health.memory.DefaultMemoryManager] 1705261465 memory requested, 0 released, 533757952 free
2020-09-30T22:52:55.622+02002020-09-30T22:52:55.622+0200 [DEBUG] [org.gradle.process.internal.health.memory.DefaultMemoryManager] 1705261465 memory requested, 559095808 free
2020-09-30T22:52:55.622+0200 [DEBUG] [org.gradle.workers.internal.WorkerDaemonExpiration] Will attempt to release 1626 of memory
2020-09-30T22:52:55.622+0200 [DEBUG] [org.gradle.process.internal.health.memory.DefaultMemoryManager] 1705261465 memory requested, 0 released, 559095808 free
 [DEBUG] [org.gradle.process.internal.health.memory.DefaultMemoryManager] 1705261465 memory requested, 559095808 free
2020-09-30T22:52:55.622+0200 [DEBUG] [org.gradle.workers.internal.WorkerDaemonExpiration] Will attempt to release 1626 of memory
2020-09-30T22:52:55.622+0200 [DEBUG] [org.gradle.process.internal.health.memory.DefaultMemoryManager] 1705261465 memory requested, 0 released, 559095808 free
2020-09-30T22:52:41.542+02002020-09-30T22:52:41.542+0200 [LIFECYCLE] [class org.gradle.internal.buildevents.TaskExecutionLogger] 
2020-09-30T22:52:41.542+0200 [LIFECYCLE [LIFECYCLE] [class org.gradle.internal.buildevents.TaskExecutionLogger] 
2020-09-30T22:52:41.542+0200 [LIFECYCLE] [class org.gradle.internal.buildevents.TaskExecutionLogger] ] [class org.gradle.internal.buildevents.TaskExecutionLogger] > Task :compileJava
> Task :compileJava
2020-09-30T22:52:57.353+02002020-09-30T22:52:57.353+0200 [LIFECYCLE] [org.gradle.cache.internal.DefaultFileLockManager] 
 [LIFECYCLE] [org.gradle.cache.internal.DefaultFileLockManager] 
2020-09-30T22:52:57.353+0200 [2020-09-30T22:52:57.353+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManagerDEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire shared lock on daemon addresses registry.
] Waiting to acquire shared lock on daemon addresses registry.
2020-09-30T22:52:57.353+0200 [DEBUG2020-09-30T22:52:57.353+0200 [DEBUG] [] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
2020-09-30T22:52:57.353+02002020-09-30T22:52:57.353+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
2020-09-30T22:52:57.353+02002020-09-30T22:52:57.353+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire shared lock on daemon addresses registry.
 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire shared lock on daemon addresses registry.
2020-09-30T22:52:57.354+0200 [DEBUG2020-09-30T22:52:57.354+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
2020-09-30T22:52:57.354+0200
2020-09-30T22:52:57.354+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
] Releasing lock on daemon addresses registry.
2020-09-30T22:52:57.142+02002020-09-30T22:52:57.142+0200 [LIFECYCLE] [class org.gradle.internal.buildevents.TaskExecutionLogger] 
 [LIFECYCLE] [class org.gradle.internal.buildevents.TaskExecutionLogger] 
2020-09-30T22:52:57.142+02002020-09-30T22:52:57.142+0200 [LIFECYCLE] [class org.gradle.internal.buildevents.TaskExecutionLogger]  [LIFECYCLE] [class org.gradle.internal.buildevents.TaskExecutionLogger] > Task :compileJava
> Task :compileJava
2020-09-30T22:53:00.624+02002020-09-30T22:53:00.624+0200 [LIFECYCLE] [org.gradle.process.internal.health.memory.DefaultMemoryManager] 
2020-09-30T22:53:00.624+0200 [DEBUG] [ [LIFECYCLE] [org.gradle.process.internal.health.memory.DefaultMemoryManager] 
2020-09-30T22:53:00.624+0200 [DEBUG] [org.gradle.process.internal.health.memory.DefaultMemoryManager] 1705261465 memory requested, 513916928 free
2020-09-30T22:53:00.624+0200org.gradle.process.internal.health.memory.DefaultMemoryManager] 1705261465 memory requested, 513916928 free
2020-09-30T22:53:00.624+0200 [DEBUG] [org.gradle.workers.internal.WorkerDaemonExpiration] Will attempt to release 1626 of memory [DEBUG] [org.gradle.workers.internal.WorkerDaemonExpiration] Will attempt to release 1626 of memory
2020-09-30T22:53:00.624+0200 [DEBUG] [
2020-09-30T22:53:00.624+0200 [DEBUG] [org.gradle.process.internal.health.memory.DefaultMemoryManager] 1705261465 memory requested, 0 released, 513916928 free
org.gradle.process.internal.health.memory.DefaultMemoryManager] 1705261465 memory requested, 0 released, 513916928 free
2020-09-30T22:53:05.624+02002020-09-30T22:53:05.624+0200 [DEBUG] [org.gradle.process.internal.health.memory.DefaultMemoryManager] 1705261465 memory requested, 489553920 free [DEBUG] [org.gradle.process.internal.health.memory.DefaultMemoryManager] 1705261465 memory requested, 489553920 free

2020-09-30T22:53:05.624+02002020-09-30T22:53:05.624+0200 [DEBUG] [org.gradle.workers.internal.WorkerDaemonExpiration]  [DEBUG] [org.gradle.workers.internal.WorkerDaemonExpiration] Will attempt to release 1626 of memoryWill attempt to release 1626 of memory

2020-09-30T22:53:05.624+02002020-09-30T22:53:05.624+0200 [DEBUG] [org.gradle.process.internal.health.memory.DefaultMemoryManager [DEBUG] [org.gradle.process.internal.health.memory.DefaultMemoryManager] 1705261465 memory requested, 0 released, 489553920 free] 1705261465 memory requested, 0 released, 489553920 free

2020-09-30T22:53:07.352+02002020-09-30T22:53:07.352+0200 [ [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire shared lock on daemon addresses registry.DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire shared lock on daemon addresses registry.

2020-09-30T22:53:07.352+02002020-09-30T22:53:07.352+0200 [ [DEBUGDEBUG] [] [org.gradle.cache.internal.DefaultFileLockManagerorg.gradle.cache.internal.DefaultFileLockManager] ] Lock acquired on daemon addresses registry.Lock acquired on daemon addresses registry.

2020-09-30T22:53:07.353+02002020-09-30T22:53:07.353+0200 [ [DEBUGDEBUG] [] [org.gradle.cache.internal.DefaultFileLockManagerorg.gradle.cache.internal.DefaultFileLockManager] ] Releasing lock on daemon addresses registry.Releasing lock on daemon addresses registry.

2020-09-30T22:53:07.354+02002020-09-30T22:53:07.354+0200 [ [DEBUGDEBUG] [] [org.gradle.cache.internal.DefaultFileLockManagerorg.gradle.cache.internal.DefaultFileLockManager] ] Waiting to acquire shared lock on daemon addresses registry.
Waiting to acquire shared lock on daemon addresses registry.
2020-09-30T22:53:07.355+02002020-09-30T22:53:07.355+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
2020-09-30T22:53:07.355+0200 [2020-09-30T22:53:07.355+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
Releasing lock on daemon addresses registry.
2020-09-30T22:52:59.347+02002020-09-30T22:52:59.347+0200 [LIFECYCLE] [class org.gradle.internal.buildevents.TaskExecutionLogger] 
 [LIFECYCLE] [class org.gradle.internal.buildevents.TaskExecutionLogger] 
2020-09-30T22:52:59.347+0200 [2020-09-30T22:52:59.347+0200 [LIFECYCLE] [class org.gradle.internal.buildevents.TaskExecutionLogger] > Task :compileJavaLIFECYCLE] [class org.gradle.internal.buildevents.TaskExecutionLogger] > Task :compileJava

2020-09-30T22:53:10.604+02002020-09-30T22:53:10.604+0200 [LIFECYCLE] [org.gradle.internal.work.DefaultWorkerLeaseService]  [LIFECYCLE] [org.gradle.internal.work.DefaultWorkerLeaseService] 

2020-09-30T22:53:10.604+02002020-09-30T22:53:10.604+0200 [ [DEBUGDEBUG] [] [org.gradle.internal.work.DefaultWorkerLeaseServiceorg.gradle.internal.work.DefaultWorkerLeaseService] ] Worker lease root.1.31.427.428 started (1 worker(s) in use).Worker lease root.1.31.427.428 started (1 worker(s) in use).

2020-09-30T22:53:10.604+02002020-09-30T22:53:10.604+0200 [DEBUG] [org.gradle.internal.resources.AbstractTrackedResourceLock]  [DEBUG] [org.gradle.internal.resources.AbstractTrackedResourceLock] Build operations Thread 7: acquired lock on root.1.31.427.428
Build operations Thread 7: acquired lock on root.1.31.427.428
2020-09-30T22:53:10.604+02002020-09-30T22:53:10.604+0200 [ [DEBUGDEBUG] [] [org.gradle.internal.work.DefaultWorkerLeaseServiceorg.gradle.internal.work.DefaultWorkerLeaseService] ] Worker lease root.1.31.427.430 started (2 worker(s) in use).
Worker lease root.1.31.427.430 started (2 worker(s) in use).
2020-09-30T22:53:10.604+02002020-09-30T22:53:10.604+0200 [DEBUG] [org.gradle.internal.resources.AbstractTrackedResourceLock] Build operations Thread 6: acquired lock on root.1.31.427.430 [DEBUG] [org.gradle.internal.resources.AbstractTrackedResourceLock] Build operations Thread 6: acquired lock on root.1.31.427.430

2020-09-30T22:53:10.604+0200 [DEBUG] [org.gradle.internal.work.DefaultWorkerLeaseService] 2020-09-30T22:53:10.604+0200 [DEBUG] [org.gradle.internal.work.DefaultWorkerLeaseService] Worker lease root.1.31.427.431 started (3 worker(s) in use).
Worker lease root.1.31.427.431 started (3 worker(s) in use).
2020-09-30T22:53:10.604+0200 [DEBUG] [2020-09-30T22:53:10.604+0200 [DEBUG] [org.gradle.internal.resources.AbstractTrackedResourceLock] Build operations Thread 8: acquired lock on root.1.31.427.431
org.gradle.internal.resources.AbstractTrackedResourceLock] Build operations Thread 8: acquired lock on root.1.31.427.431
2020-09-30T22:53:10.605+0200 [2020-09-30T22:53:10.605+0200 [DEBUG] [org.gradle.internal.work.DefaultWorkerLeaseService] Worker lease root.1.31.427.429 started (4 worker(s) in use).DEBUG] [org.gradle.internal.work.DefaultWorkerLeaseService] Worker lease root.1.31.427.429 started (4 worker(s) in use).
2020-09-30T22:53:10.605+0200
2020-09-30T22:53:10.605+0200 [DEBUG] [org.gradle.internal.resources.AbstractTrackedResourceLock] Build operations: acquired lock on root.1.31.427.429 [DEBUG] [org.gradle.internal.resources.AbstractTrackedResourceLock] Build operations: acquired lock on root.1.31.427.429
2020-09-30T22:53:10.605+0200
2020-09-30T22:53:10.605+0200 [DEBUG] [org.gradle.internal.work.DefaultWorkerLeaseService [DEBUG] [org.gradle.internal.work.DefaultWorkerLeaseService] Worker lease root.1.31.427.436 started (5 worker(s) in use).
] Worker lease root.1.31.427.436 started (5 worker(s) in use).
2020-09-30T22:53:10.605+0200 [DEBUG2020-09-30T22:53:10.605+0200 [DEBUG] [org.gradle.internal.resources.AbstractTrackedResourceLock] ] [org.gradle.internal.resources.AbstractTrackedResourceLock] Build operations Thread 4: acquired lock on root.1.31.427.436
Build operations Thread 4: acquired lock on root.1.31.427.436
```

**Normal version (Gradle 6.4.1):**

![image](https://user-images.githubusercontent.com/14804545/94796520-31060300-03df-11eb-8b47-826e646030de.png)

```
2020-09-30T22:54:09.291+0200 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationExecutor2020-09-30T22:54:09.291+0200 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationExecutor] Build operation 'Compile Java for :compileJava' started] Build operation 'Compile Java for :compileJava' started

2020-09-30T22:54:09.337+02002020-09-30T22:54:09.337+0200 [INFO [INFO] [org.gradle.api.internal.tasks.compile.JdkJavaCompiler] [org.gradle.api.internal.tasks.compile.JdkJavaCompiler] Compiling with JDK Java compiler API.] Compiling with JDK Java compiler API.

2020-09-30T22:54:16.848+02002020-09-30T22:54:16.848+0200 [LIFECYCLE] [org.apache.http.impl.conn.PoolingHttpClientConnectionManager]  [LIFECYCLE] [org.apache.http.impl.conn.PoolingHttpClientConnectionManager] 

2020-09-30T22:54:16.848+02002020-09-30T22:54:16.848+0200 [DEBUG] [ [DEBUG] [org.apache.http.impl.conn.PoolingHttpClientConnectionManager] Connection manager is shutting down
org.apache.http.impl.conn.PoolingHttpClientConnectionManager] Connection manager is shutting down
2020-09-30T22:54:16.849+02002020-09-30T22:54:16.849+0200 [DEBUG] [org.apache.http.impl.conn.DefaultManagedHttpClientConnection] http-outgoing-6: Close connection
 [DEBUG] [org.apache.http.impl.conn.DefaultManagedHttpClientConnection] http-outgoing-6: Close connection
2020-09-30T22:54:16.849+02002020-09-30T22:54:16.849+0200 [ [DEBUG] [DEBUG] [org.apache.http.impl.conn.PoolingHttpClientConnectionManagerorg.apache.http.impl.conn.PoolingHttpClientConnectionManager] Connection manager shut down] Connection manager shut down

2020-09-30T22:54:16.851+02002020-09-30T22:54:16.851+0200 [DEBUG] [ [DEBUG] [org.apache.http.impl.conn.PoolingHttpClientConnectionManager] Connection manager is shutting down
org.apache.http.impl.conn.PoolingHttpClientConnectionManager] Connection manager is shutting down
2020-09-30T22:54:16.851+0200 [DEBUG2020-09-30T22:54:16.851+0200 [DEBUG] [] [org.apache.http.impl.conn.DefaultManagedHttpClientConnection] org.apache.http.impl.conn.DefaultManagedHttpClientConnection] http-outgoing-8: Close connection
http-outgoing-8: Close connection
2020-09-30T22:54:16.852+02002020-09-30T22:54:16.852+0200 [DEBUG] [org.apache.http.impl.conn.PoolingHttpClientConnectionManager]  [DEBUG] [org.apache.http.impl.conn.PoolingHttpClientConnectionManager] Connection manager shut down
Connection manager shut down
2020-09-30T22:54:17.427+02002020-09-30T22:54:17.427+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire shared lock on daemon addresses registry.
2020-09-30T22:54:17.427+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire shared lock on daemon addresses registry.
2020-09-30T22:54:17.427+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager]  [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
Lock acquired on daemon addresses registry.
2020-09-30T22:54:17.427+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] 2020-09-30T22:54:17.427+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
Releasing lock on daemon addresses registry.
2020-09-30T22:54:17.427+02002020-09-30T22:54:17.427+0200 [DEBUG] [ [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire shared lock on daemon addresses registry.Waiting to acquire shared lock on daemon addresses registry.
2020-09-30T22:54:17.427+0200 [
2020-09-30T22:54:17.427+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.

2020-09-30T22:54:17.427+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager2020-09-30T22:54:17.427+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
] Releasing lock on daemon addresses registry.
2020-09-30T22:54:15.917+02002020-09-30T22:54:15.917+0200 [LIFECYCLE] [class org.gradle.internal.buildevents.TaskExecutionLogger] 
 [LIFECYCLE] [class org.gradle.internal.buildevents.TaskExecutionLogger] 
2020-09-30T22:54:15.917+02002020-09-30T22:54:15.917+0200 [LIFECYCLE] [class org.gradle.internal.buildevents.TaskExecutionLogger] > Task :compileJava
 [LIFECYCLE] [class org.gradle.internal.buildevents.TaskExecutionLogger] > Task :compileJava
2020-09-30T22:54:19.671+02002020-09-30T22:54:19.671+0200 [LIFECYCLE] [org.gradle.internal.work.DefaultWorkerLeaseService] 
2020-09-30T22:54:19.671+0200 [ [LIFECYCLE] [org.gradle.internal.work.DefaultWorkerLeaseService] 
2020-09-30T22:54:19.671+0200 [DEBUG] [DEBUG] [org.gradle.internal.work.DefaultWorkerLeaseService] org.gradle.internal.work.DefaultWorkerLeaseService] Worker lease root.1.24.427.428 started (1 worker(s) in use).
Worker lease root.1.24.427.428 started (1 worker(s) in use).
2020-09-30T22:54:19.671+02002020-09-30T22:54:19.671+0200 [DEBUG] [org.gradle.internal.resources.AbstractTrackedResourceLock] Build operations: acquired lock on root.1.24.427.428 [DEBUG] [org.gradle.internal.resources.AbstractTrackedResourceLock] Build operations: acquired lock on root.1.24.427.428

2020-09-30T22:54:19.671+0200 [2020-09-30T22:54:19.671+0200 [DEBUG] [org.gradle.internal.work.DefaultWorkerLeaseServiceDEBUG] [org.gradle.internal.work.DefaultWorkerLeaseService] Worker lease root.1.24.427.436 started (2 worker(s) in use).
] Worker lease root.1.24.427.436 started (2 worker(s) in use).
2020-09-30T22:54:19.671+02002020-09-30T22:54:19.671+0200 [DEBUG] [ [DEBUG] [org.gradle.internal.resources.AbstractTrackedResourceLock] Build operations Thread 8: acquired lock on root.1.24.427.436org.gradle.internal.resources.AbstractTrackedResourceLock] Build operations Thread 8: acquired lock on root.1.24.427.436

2020-09-30T22:54:19.671+02002020-09-30T22:54:19.671+0200 [DEBUG] [ [DEBUG] [org.gradle.internal.work.DefaultWorkerLeaseService] Worker lease root.1.24.427.435 started (3 worker(s) in use).org.gradle.internal.work.DefaultWorkerLeaseService] Worker lease root.1.24.427.435 started (3 worker(s) in use).

2020-09-30T22:54:19.671+0200 [2020-09-30T22:54:19.671+0200 [DEBUG] [org.gradle.internal.resources.AbstractTrackedResourceLockDEBUG] [org.gradle.internal.resources.AbstractTrackedResourceLock] Build operations Thread 6: acquired lock on root.1.24.427.435
] Build operations Thread 6: acquired lock on root.1.24.427.435
2020-09-30T22:54:19.672+0200 [2020-09-30T22:54:19.672+0200 [DEBUG] [org.gradle.internal.work.DefaultWorkerLeaseServiceDEBUG] [org.gradle.internal.work.DefaultWorkerLeaseService] Worker lease root.1.24.427.434 started (4 worker(s) in use).
] Worker lease root.1.24.427.434 started (4 worker(s) in use).
2020-09-30T22:54:19.672+02002020-09-30T22:54:19.672+0200 [DEBUG] [org.gradle.internal.resources.AbstractTrackedResourceLock] Build operations Thread 7: acquired lock on root.1.24.427.434
 [DEBUG] [org.gradle.internal.resources.AbstractTrackedResourceLock] Build operations Thread 7: acquired lock on root.1.24.427.434
2020-09-30T22:54:19.672+02002020-09-30T22:54:19.672+0200 [DEBUG] [ [DEBUG] [org.gradle.internal.work.DefaultWorkerLeaseService] org.gradle.internal.work.DefaultWorkerLeaseService] Worker lease root.1.24.427.433 started (5 worker(s) in use).
Worker lease root.1.24.427.433 started (5 worker(s) in use).
2020-09-30T22:54:19.672+02002020-09-30T22:54:19.672+0200 [DEBUG] [org.gradle.internal.resources.AbstractTrackedResourceLock] Build operations Thread 4: acquired lock on root.1.24.427.433
 [DEBUG] [org.gradle.internal.resources.AbstractTrackedResourceLock] Build operations Thread 4: acquired lock on root.1.24.427.433
```

### Steps to Reproduce 
The demo project has some artificial complexity (dependencies and duplicated classes) added to it to achieve visible differences between different versions (in seconds instead of milliseconds).
`git clone https://github.com/ve2m/gradle-regression-demo.git`

`./gradlew wrapper --gradle-version=6.4.1`
Tru to run this 3-5 times and note how much time it takes to execute `compileJava` (~3.2 secs for me).
`./gradlew clean compileJava --info`

`./gradlew wrapper --gradle-version=6.6.1`
Tru to run this 3-5 times and note that it takes more time to execute `compileJava` (~4.2 secs for me).
`./gradlew clean compileJava --info`

### Your Environment

**Operating system** - Windows 10 10.0
**CPU cores** - 8 cores  
**Max Gradle workers** - 8 workers
**Java runtime** - Oracle OpenJDK Runtime Environment 11.0.2+9
**Java VM** - Oracle OpenJDK 64-Bit Server VM 11.0.2+9 (mixed mode)
**Max JVM memory heap size** - 2147 MB

**Comment by ve2m** [comment]
Removing some annotation processors (`hibernate-jpamodelgen`, `spring-boot-configuration-processor`) makes it much faster (~39s to ~16s for our project) but it's still worse than ~12s when using 6.0-6.4.1.

**Comment by ve2m** [comment]
Closed in favor of #14877 with a more accurate description of the problem and build scans.
